core/cli/src/plugin.ts

```diff
@@ -41,6 +41,16 @@ export async function loadPlugin(
     children: [] as Plugin[]
   }
 
+  if (!isAppRoot && plugin.rcFile.version !== CURRENT_RC_FILE_VERSION) {
+    return invalid([
+      `plugin ${s.plugin(id)} has a v${s.code((plugin.rcFile.version ?? 1).toString())} ${s.code(
+        '.toolkitrc.yml'
+      )}, but this version of Tool Kit can only load v${s.code(
+        CURRENT_RC_FILE_VERSION.toString()
+      )}. please update this plugin.`
+    ])
```

Comment:
**thought:** Some assisting documentation to add as part of the major release to explain why this was done.

Reply:
yep. documentation (including a migration guide) will be coming later.